metric_learn/base_metric.py

```diff
@@ -178,8 +214,57 @@ def transform(self, X):
                              accept_sparse=True)
     return X_checked.dot(self.transformer_.T)
 
-  def metric(self):
-    return self.transformer_.T.dot(self.transformer_)
+  def get_metric(self):
+    """Returns a function that returns the learned metric between two points.
+    This function will be independent from the metric learner that learned it
+    (it will not be modified if the initial metric learner is modified).
+
+    Returns
+    -------
+    metric_fun : function
+      The function described above.
+
+    See Also
+    --------
+    score_pairs : a method that returns the metric between several pairs of
+      points. But this is a method of the metric learner and therefore can
+      change if the metric learner changes. Besides, it can use the metric
+      learner's preprocessor, and works on concatenated arrays.
+
+    :ref:`mahalanobis_distances` : The section of the project documentation
+      that describes Mahalanobis Distances.
+    """
+    transformer_T = self.transformer_.T.copy()
+
+    def metric_fun(u, v):
+      """This function computes the metric between u and v, according to the
+      previously learned metric.
+
+      Parameters
+      ----------
+      u : array-like, shape=(n_features,)
+        The first point involved in the distances computation.
+      v : array-like, shape=(n_features,)
+        The second point involved in the distances computation.
+      Returns
+      -------
+      distance: float
+        The distance between u and v according to the new metric.
+      """
+      u = _validate_vector(u)
+      v = _validate_vector(v)
```

Comment:
Here I use scipy's _validate_vector function (used in functions like `scipy.spatial.distance.euclidean`), to more easily mimic scipy's distances behaviour (regarding 1D arrays etc). But I see the underscore at the beginning, which would mean that this function is private. Does it mean I am not supposed to use it ?

Reply:
Understood, I'll replace it by something else